### PR TITLE
BAEL-4396: Find the last modified file in a directory with Java

### PR DESCRIPTION
**Pull request by azhwani**
Simple PR

### CODE REVIEW

**Comment by doodoroma** [comment]
Retest this please

File: core-java-modules/core-java-io-3/src/test/java/com/baeldung/lastmodifiedfile/LastModifiedFileAppUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        if (!srcDir.exists())
+            srcDir.mkdir();
+
+        FileUtils.cleanDirectory(srcDir);
+
+    }
+
+    @Test
+    public void givenDirectory_whenUsingIoApi_thenFindLastModfile() throws IOException {
+        Path file01 = Paths.get(SOURCEDIRECTORY + "/file01.txt");
```
Comment (doodoroma) [inline core-java-modules/core-java-io-3/src/test/java/com/baeldung/lastmodifiedfile/LastModifiedFileAppUnitTest.java:31]: @azhwani I know I am caviling here, but the first test is testing against 1 file only, so we don't really test anything, the first modified file would also pass :)

Comment (azhwani) [reply core-java-modules/core-java-io-3/src/test/java/com/baeldung/lastmodifiedfile/LastModifiedFileAppUnitTest.java:34]: @doodoroma, I have made some changes, can you please take a look ;)

Comment (doodoroma) [reply core-java-modules/core-java-io-3/src/test/java/com/baeldung/lastmodifiedfile/LastModifiedFileAppUnitTest.java:34]: It was not clear, but the situation is true for all tests, in fact. Unit tests must be independent from each other. Maybe a small fixture could be added?

File: core-java-modules/core-java-io-3/src/main/java/com/baeldung/lastmodifiedfile/LastModifiedFileApp.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    public static File findUsingIOApi(String sdir) {
+        File lastModFile = null;
+        File dir = new File(sdir);
+        if (dir.isDirectory()) {
+            Optional<File> opFile = Arrays.stream(dir.listFiles(File::isFile))
+                .max((f1, f2) -> Long.compare(f1.lastModified(), f2.lastModified()));
+
+            if (opFile.isPresent())
+                return opFile.get();
```
Comment (eric-martin) [inline core-java-modules/core-java-io-3/src/main/java/com/baeldung/lastmodifiedfile/LastModifiedFileApp.java:25]: need braces

File: core-java-modules/core-java-io-3/src/main/java/com/baeldung/lastmodifiedfile/LastModifiedFileApp.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        return lastModFile;
+    }
+
+    public static Path findUsingNIOApi(String sdir) throws IOException {
+        Path lastModPath = null;
+        Path dir = Paths.get(sdir);
+        if (Files.isDirectory(dir)) {
+            Optional<Path> opPath = Files.list(dir)
+                .filter(p -> !Files.isDirectory(p))
```
Comment (eric-martin) [inline core-java-modules/core-java-io-3/src/main/java/com/baeldung/lastmodifiedfile/LastModifiedFileApp.java:37]: continuations are indented two spaces

**Pull request closed without merging**